actioncable/lib/action_cable/subscription_adapter/redis.rb

```diff
@@ -17,15 +18,15 @@ def initialize(*)
       end
 
       def broadcast(channel, payload)
-        redis_connection_for_broadcasts.publish(channel, payload)
+        redis_connection_for_broadcasts.publish(channel_name(channel), payload)
```

Comment:
Instead of making the adapter call `channel_name`, if we `prepend` the module, it could define `broadcast`/`subscribe`/`unsubscribe` methods that mangle the name then `super`.